src/lib.zig

```diff
@@ -246,7 +246,10 @@ export fn gemtextDocumentRemove(document: *c.gemtext_document, index: usize) voi
         );
     }
 
-    fragments = allocator.shrink(fragments, fragments.len - 1);
+    fragments = allocator.realloc(fragments, fragments.len - 1) catch |err| t: {
+        std.log.warn("could not resize fragments: {}", .{err});
+        break :t fragments;
+    };
```

Comment:
So I took a look at the allocators, but I'm getting the impression that `shrink` may be no more.  There's `resize`, but when I tried using it here, the slice len was not updated, which caused bad things :tm: 

If you have a better suggestion I'm more than happy to change this to something better.

Reply:
If resize fails, you can do a malloc and free to create a new slice and basically implement `realloc` like it behaves in C

Reply:
So the odd thing is that my original use was with `resize`, and I did indeed check the return for a fail.  The return of `resize` returned `true`, however the slice len remained `3`.  The discriminator of the _last_ fragment after the `memcpy` seemed corrupt, too, which is what I was getting at with the bad things :tm:. 

Let me know what you would prefer, I can give a shot at what you're talking about too!